src/components/CourseForm/index.test.tsx

```diff
@@ -401,4 +688,123 @@ describe('Course Form Course Edit Tests', () => {
+      description: 'A test course which has been edited',
+      summary: 'Updated version of the course',
+      startDate: courseStart,
+      endDate: courseEnd,
+      lastEnrollDate: oneDayBeforeStart,
+      lastCancelDate: oneDayBeforeStart,
+      maxStudents: 40,
+      tags: [{ id: '2', name: 'Git' }],
+      image: 'http://test-image.com',
+    };
+
+    renderWithTheme(
+      <CourseForm lang="en" tags={tags} courseData={course} templates={[]} />
+    );
+
+    const name = screen.getByTestId('courseFormName') as HTMLInputElement;
+    await user.clear(name);
+    await user.type(name, editedCourse.name);
+
+    const description = screen.getByTestId('courseFormDescription');
+    await user.type(description, ' which has been edited');
+
+    const summary = screen.getByTestId('courseFormSummary') as HTMLInputElement;
+    await user.clear(summary);
+    await user.type(summary, editedCourse.summary);
+
+    const maxStudents = screen.getByTestId(
+      'courseFormMaxStudents'
+    ) as HTMLInputElement;
+    await user.clear(maxStudents);
+    await user.type(maxStudents, editedCourse.maxStudents.toString());
+
+    const image = screen.getByTestId('courseFormImage') as HTMLInputElement;
+    await user.type(image, editedCourse.image);
+
+    const tagsInput = screen.getByRole('combobox', {
+      name: /CourseForm.tags/i,
+    });
+    await user.click(tagsInput);
+    const option = screen.getByText(tags[1].name);
+    await user.click(option);
+
+    const submitButton = screen.getByTestId('courseFormSubmit');
+    await userEvent.click(submitButton);
+
+    await waitFor(() => {
+      expect(mockFetch).toBeCalledWith('/api/course', {
+        ...editedCourse,
+        endDate: new Date(course.endDate),
+        startDate: new Date(course.startDate),
+        lastEnrollDate: new Date(course.lastEnrollDate),
+        lastCancelDate: new Date(course.lastCancelDate),
+        maxStudents: Number(editedCourse.maxStudents),
+        tags: [tags[1].name],
+      });
+    });
+  });
+
+  it('should not be possible to clear name field and submit', async () => {
+    mockFetch.mockClear();
+    const user = userEvent.setup();
+
+    const course = {
+      id: '1234',
+      createdById: '30',
+      name: 'New course',
+      description: 'A test course',
+      summary: 'All you ever wanted to know about testing!',
+      startDate: courseStart,
+      endDate: courseEnd,
+      lastEnrollDate: oneDayBeforeStart,
+      lastCancelDate: oneDayBeforeStart,
+      maxStudents: 55,
+      tags: [],
+      image: '',
+    };
+
+    renderWithTheme(
+      <CourseForm lang="en" tags={[]} courseData={course} templates={[]} />
+    );
+
+    const name = screen.getByTestId('courseFormName') as HTMLInputElement;
+    await user.clear(name);
+
+    const submitButton = screen.getByTestId('courseFormSubmit');
+    await userEvent.click(submitButton);
+
+    await waitFor(() => {
+      expect(mockFetch).not.toBeCalled();
+    });
+  });
```

Comment:
Description field is also required field so maybe create a similar test to it as the above test?  it('should not be possible to clear description field and submit').

Reply:
This test is not doable at the moment. There is a bug when editing a course. The edit form can be submitted even though the description field is empty. We will create an issue that will fix this.